scripts/xsslint/xsslint/linters.py

```diff
@@ -457,15 +458,13 @@ def _check_javascript_escape(self, file_contents, results):
         """
         Checks that only necessary escape() are used.
 
-        Allows for _.escape(), although this shouldn't be the recommendation.
-
         Arguments:
             file_contents: The contents of the JavaScript file.
             results: A file results objects to which violations will be added.
 
         """
-        # Ignores calls starting with "_.", because those are safe
-        regex = regex = re.compile(r"(?<!_).escape\(")
+        # Ensure that calls to fooescape or foo.escape are ignored; they aren't the dangerous escape functions.
+        regex = re.compile(r"(?:^|(?<![\w.$]))escape\(")
```

Comment:
@robrap: Could you please take a look at this adjustment to the XSS linting? It seems reasonable to me, but you have a lot more context on this than I do.